### ISSUE TITLE
Enable vertex colors as albedo in the fallback material (but not in StandardMaterial3D)

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot-proposals#3031]
### Describe the project you are working on

The Godot editor :slightly_smiling_face: 

### Describe the problem or limitation you are having in your project

Vertex colors cannot be previewed immediately on 3D meshes. You have to create a StandardMaterial3D resource, enable **Vertex Color > Use As Albedo** on the material and assign it to the mesh. This tends to be a usability hurdle when performing procedural geometry generation that relies on vertex color ([here's an example of this happening in the wild](https://godotforums.org/discussion/25198/i-need-explanation-on-arraymesh-trying-to-create-a-simple-cube)).

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Enable **Vertex Color > Use As Albedo** on the fallback 3D material, but keep StandardMaterial3D/ORMMaterial3D as they are.

By "fallback material", I'm referring to the material that is displayed when no material is assigned to a mesh. This material looks slightly different to the default StandardMaterial3D in the `master` branch (it's darker).

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Modify the fallback material code to interpret vertex color as an albedo multiplier.

**Question: Why not enable vertex color as albedo in StandardMaterial3D as well?**

Vertex color has a cost, especially on complex meshes. It should only be enabled when you actually need it.

However, it is unlikely that you'll be using the fallback material anywhere in a finished project. Therefore, the potential performance impact is not as relevant if it only impacts the fallback material, not the default StandardMaterial3D.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No, as this is about improving the engine usability for procedural geometry generation and related tools (such as editor plugins that may rely on it).

### Is there a reason why this should be core and not an add-on in the asset library?

See above.

**Comment by clayjohn** [comment]
I'm not sure if this is a good idea. I think we need to wait and see if there is an actual demand from users. While it would be handy for visualizing certain meshes, many other meshes use vertex color as a general purpose way of passing 4 values in to each vertex. It is very commonly used for blending of mesh properties.

**Comment by Calinou** [comment]
> It is very commonly used for blending of mesh properties.

In this case, you'll be using a custom material anyway :slightly_smiling_face:

**Comment by Calinou** [comment]
Closing, as https://github.com/godotengine/godot/pull/50830 was rejected.

**Comment by insomniacUNDERSCORElemon** [comment]
Is there a way to set this manually? Even some convoluted name-it-manually way to create the fallback material?

I'm coming from the opposite workflow: I want to use vertex colors for most-if-not-all models (no textures). I can do this with glTF\* but it also means that every import will have its own material(s) which is likely its own performance issue. Most things could use the same material (maybe a few other general-purpose ones, and full-object color can easily be done in-engine instead).

The only other thing I could think of is if there is some way to auto-merge/override materials based on name (would be nice if the importer handled it), or if all models could be done in a single import which would be converted into a mesh library (better if usable outside of a gridmap).

![planter_low](https://github.com/godotengine/godot-proposals/assets/4397725/202007a5-7e6c-4a17-9586-833103187082)

\*= obj seems to not work with vertex colors in 3.X, which I am locked to

(sidenote: Blender allows multiple vertex color maps and inputting these into materials, for instance using it for glow... but this seems to break in Godot 3.X)

**Comment by Calinou** [comment]
@insomniacUNDERSCORElemon There's no way to override the fallback material, but this wouldn't address your use case as your imported 3D scenes *are* using materials.

It sounds like [shader templates](https://github.com/godotengine/godot-proposals/issues/8366) could be a solution here.

**Comment by insomniacUNDERSCORElemon** [comment]
> enable **Use External** for the material in the Advanced Import Settings

This sounds like the closest thing to an answer, if only I could use 4.X with my desired language (that's also one reason why I'm trying textureless 3D in the first place). I'd hope this would work with presets (easy to import groups by normal/shiny etc), though I could see objects that use multiple materials being a potential complication.

--------------

On multiple vertex color maps, I edited my comment above but I tested it and the fault is actually during export (not saving the 2nd map, at least w/this format). Also I'd wonder if color indexing could be helpful with performance (and maybe allow tweaking), though likely only useful for already low-color. 

I could see maybe using unneeded data for extra shading but I'd probably try using vertex lighting for that first (also not in 4.X but I do see the PR to fix it). Though likely the only thing I'd do here with a custom shader is using pure green etc for effect placement.
